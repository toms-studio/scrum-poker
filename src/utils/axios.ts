import axios, { AxiosRequestConfig } from "axios";

const customAxios = axios.create({
  baseURL: process.env.NEXT_PUBLIC_API_URL,
});

customAxios.interceptors.request.use((config: AxiosRequestConfig) => {
  // custom token or headers here...
  return config;
});

export default customAxios;
