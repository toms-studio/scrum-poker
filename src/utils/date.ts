import dayjs from "dayjs";

const DEFAULT_FORMAT = "MM-DD-YYYY";

// List available token format https://day.js.org/docs/en/parse/string-format#list-of-all-available-parsing-tokens
function dateFormat(date: string, format: string = DEFAULT_FORMAT): string {
  return dayjs(date).format(format);
}

function today(format: string = DEFAULT_FORMAT): string {
  return dayjs().format(format);
}

export { dateFormat, today };
