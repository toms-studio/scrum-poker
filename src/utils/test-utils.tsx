import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { RenderOptions, render } from "@testing-library/react";
import React, { FC, ReactElement } from "react";
import { Provider, createStore } from "~/store";

const queryClient = new QueryClient({});

const ReactQueryProvider: FC = ({ children }) => {
  return (
    <QueryClientProvider client={queryClient}>
      <Provider createStore={createStore}>{children}</Provider>
    </QueryClientProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: ReactQueryProvider, ...options });

export * from "@testing-library/react";
export { customRender as render };
