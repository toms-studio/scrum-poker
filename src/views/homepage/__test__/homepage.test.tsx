import React from "react";

import { render, screen } from "~/utils/test-utils";

import Homepage from "../homepage";

describe("<Homepage />", () => {
  it("renders a homepage", () => {
    render(<Homepage />);

    const heading = screen.getByText("VirtualSpirit");

    expect(heading).toBeInTheDocument();
    expect(heading).toMatchSnapshot();
  });
});
