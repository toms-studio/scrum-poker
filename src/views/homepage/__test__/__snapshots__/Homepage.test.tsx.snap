// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Homepage /> renders a homepage 1`] = `
<h1
  class="text-4xl font-bold tracking-tight sm:text-center sm:text-6xl"
>
  VirtualSpirit
</h1>
`;
